pkg/registry/registry.go

```diff
@@ -336,18 +335,15 @@ func (r *Registry) handleBlob(rw mux.ResponseWriter, req *http.Request, ref refe
 	if req.Method == http.MethodHead {
 		return
 	}
-	var w io.Writer = rw
+	var w mux.ResponseWriter = rw
 	rc, err := r.ociClient.GetBlob(req.Context(), ref.dgst)
 	if err != nil {
 		rw.WriteError(http.StatusInternalServerError, fmt.Errorf("could not get reader for blob with digest %s: %w", ref.dgst.String(), err))
 		return
 	}
 	defer rc.Close()
-	_, err = io.Copy(w, rc)
-	if err != nil {
-		r.log.Error(err, "error occurred when copying blob")
-		return
-	}
+
+	http.ServeContent(w, req, ref.dgst.String(), time.Time{}, rc)
```

Comment:
I remember there being a reason I did not use `ServeContent` initially. Let me look up if the reason I found still holds.

Reply:
Size no longer needs to be set at the beginning of the function as it can now be determined by seek. Also set an empty string as the name.